src/kotlin_language.cpp

```diff
@@ -364,17 +363,20 @@ bool KotlinLanguage::handles_global_class_type(const String& p_type) const {
 String KotlinLanguage::get_global_class_name(const String& p_path, String* r_base_type, String* r_icon_path) const {
     if (p_path.begins_with(GODOT_ENTRY_PATH)) { return String(); }
 
-    if (KtClass* clazz {GDKotlin::get_instance().find_class(p_path)}) {
-        if (r_base_type) { *r_base_type = clazz->base_godot_class; }
-
-        return clazz->registered_class_name;
+    String script_name = p_path.get_file().trim_suffix(p_path.get_extension()).trim_suffix(".");
```

Comment:
scriptNameExtraction: I see this 3 times (maybe there are more?) in the code at different locations. I would move that to a separate function somewhere so this scriptName extraction is handled in one place in case we need to change it later on.